src/Nox.Generator/Common/TemplateCodeBuilder.cs

```diff
@@ -136,10 +150,23 @@ private void GenerateSourceCode(string template, object model, string sourceFile
         NoxSolutionBridge.AddFunctions(context, _codeGeneratorState.Solution);
 
         var sourceText = strongTemplate.Render(context);
-        
-        _context.AddSource(sourceFileName,
-            SourceText.From(sourceText,
-            Encoding.UTF8));
+
+        if (string.IsNullOrEmpty(_fileExtension) || _fileExtension == "cs")
+        {
+            _context.AddSource($"{sourceFileName}.g.cs", SourceText.From(sourceText, Encoding.UTF8));
+        }
+        else
+        {
```

Comment:
This is a very rough implementation. We should probably configure output path on the client side, maybe in appsetings.json

Reply:
@anna-naumova25  lets think about composition or inheritance to solve this, so we keep 2 generators the simple and with only the specific code.

Reply:
I've create TemplateCodeBuilderBase and two derived classes - TemplateCodeBuilder and TemplateCodeBuilderFile to separate the logic